mlflow/sklearn/__init__.py

```diff
@@ -1437,18 +1437,19 @@ def get_input_example():
                 return input_example
 
         def infer_model_signature(input_example):
-            if not hasattr(estimator, "predict"):
+            if hasattr(estimator, "predict"):
+                # Copy the input example so that it is not mutated by the call to
+                # predict() prior to signature inference
+                model_output = estimator.predict(deepcopy(input_example))
+            elif hasattr(estimator, "transform"):
+                model_output = estimator.transform(deepcopy(input_example))
+            else:
                 raise Exception(
-                    "the trained model does not specify a `predict` function, "
-                    + "which is required in order to infer the signature"
+                    "the trained model does not specify a `predict` or `transform` "
```

Comment:
```suggestion
                    "the trained model does not have a `predict` or `transform` "
```

`specify` sounds a bit strange.